docs/parthenon_arrays.md

```diff
@@ -0,0 +1,88 @@
+# Parthenon Arbitrary-Dimensional Arrays
+
+`ParArrayND` is a wrapper around a rank 6 `Kokkos::View`. It provides
+a rank-agnostic way to create, managed, and carry
+`Kokkos::Views`. Since it is built on `Kokkos::View`, it is reference
+counted, works on GPUs, and is almost as performant as `Kokkos::View`.
+
+The index and size convention is the same as in `Athena++`. The
+fastest moving indexes come last.
+
+### Constructors
+
+To construct, call it with `ParArrayND(label,dimensions...)`, where
+dimensions is some number of ints specifying the shape of the array.
+
+If you don't know what to name your array, try using the
+`PARARRAY_TEMP` macro. It is provided to name the array based on the
+file and line where it is created.
```

Comment:
Yep, exactly. I tried overloading a constructor, and hit the problem that the name always  pointed to the line in `parthenon_arrays.hpp` where the constructor is defined. I also tried wrapping the constructor with its own macro... but given all the templates, this ended up with some pretty nasty, unintuitive syntax. A macro for the name seemed like the least bad option.

Reply:
@jlippuner I can't reply to your comment below, so I'll just quote it here:

> Hmm. I'm a bit torn about this too. Unlike Kokkos, I guess we require a label because it needs to be indexed in the container, is that right?

I designed `ParArrayND` to require a label because `Kokkos` requires one and providing one ensures that `kokkos-tools` provide more useful information. It's a bit weirdly redundant to provide a label here and in `Container`. But I think it's valuable to interface with `Kokkos` this way.

> Maybe the best solution would be to just require the user to provide a label. They can always write foo ;)

That was how I originally designed it. :) @jdolence has been beta testing this in his refactor branch, though, and we immediately hit the situation where debugging a bunch of random temporary arrays named `foo` or `temp1`, `temp2`, etc., is annoying. I can get rid of this if you feel strongly, but I think it provides some value.